activerecord/lib/active_record/explain.rb

```diff
@@ -50,5 +50,21 @@ def render_bind(attr)
 
         [attr&.name, value]
       end
+
+      def build_explain_clause(options = [])
+        if connection.respond_to?(:build_explain_clause)
+          connection.build_explain_clause(options)
+        else
+          "EXPLAIN for:"
+        end
+      end
+
+      def connection_explain(sql, binds, options)
+        if connection.method(:explain).parameters.size == 2
+          connection.explain(sql, binds)
```

Comment:
Same here about the deprecation.

Reply:
I added the deprecation here but not above since it would result in a double deprecation statement. Let me know if this is good